docs/src/man/comparisons.md

```diff
@@ -1,77 +1,204 @@
+|                           | `df[['x','y']].mean()`                                                       | `combine(df, [:x, :y] .=> mean)`                              |
+|                           | `df.filter(regex=("^x")).mean()`                                             | `combine(df, r"^x" .=> mean)`                                 |
+|                           | `df[['x', 'y']].agg([max, min])`                                             | `combine(df, ([:x, :y] .=> [maximum minimum])...)`            |
+| Multivariate function     | `df.assign(x_y_cor = np.corrcoef(df.x, df.y)[0,1])`                          | `transform(df, [:x, :y] => cor)`                              |
+| Row-wise                  | `df.assign(x_y_min = df.apply(lambda v: min(v.x, v.y), axis=1))`             | `transform(df, [:x, :y] => ByRow(min))`                       |
+|                           | `df.assign(x_y_argmax = df.apply(lambda v: df.columns[v.argmax()], axis=1))` | `transform(df, AsTable([:x,:y]) => ByRow(argmax))`            |
+| DataFrame as input        | `df.groupby('grp').head(2)`                                                  | `combine(d -> first(d, 2), groupby(df, :grp))`                |
+| DataFrame as output       | `df[['x']].agg(['min','max'])`                                               | `combine(:x => x -> (value = [minimum(x), maximum(x)],), df)` |
+
+### Joining data frames
+
+Suppose that you have a second data frame as shown below:
+```
+df2 = pandas.DataFrame({'grp': [1], 'w': [10]})
+
+>>> df2
+   grp   w
+0    1  10
+```
+
+Here is how to join the data frames:
+
+| Operation             | pandas                                         | DataFrames.jl                   |
+| :-------------------- | :--------------------------------------------- | :------------------------------ |
+| Inner join            | `pd.merge(df, df2, how = 'inner', on = 'grp')` | `innerjoin(df, df2, on = :grp)` |
+| Outer join            | `pd.merge(df, df2, how = 'outer', on = 'grp')` | `outerjoin(df, df2, on = :grp)` |
+| Left join             | `pd.merge(df, df2, how = 'left', on = 'grp')`  | `leftjoin(df, df2, on = :grp)`  |
+| Right join            | `pd.merge(df, df2, how = 'right', on = 'grp')` | `rightjoin(df, df2, on = :grp)` |
+| Semi join (filtering) | `df[df.grp.isin(df2.grp)]`                     | `semijoin(df, df2, on = :grp)`  |
+| Anti join (filtering) | `df[~df.grp.isin(df2.grp)]`                    | `antijoin(df, df2, on = :grp)`  |
+
+### Additional Differences
+
+1. Pandas skips `NaN` values in analytic functions by default. In Julia `NaN` is just a normal floating point value, and instead a special `missing` value is used to indicate missing data. DataFrames.jl respects general rules in Julia in propagating `missing` values by default.
+
+2. Pandas keeps original column name after performing an analytic function. DataFrames.jl appends a suffix to the column name by default.
```

Comment:
agreed - still we should then give one example how to achieve the non-default results in both frameworks as commented above.

Reply:
Added new example. Please review this commit  https://github.com/JuliaData/DataFrames.jl/pull/2378/commits/4cf28eccc5a542d93f2c8e3c53c19eeb69c4c52a.